test/test_ops.py

```diff
@@ -981,6 +992,37 @@ def _case_four_transform(t):
                     with self.assertRaises(RuntimeError, msg=msg_fail):
                         op_out(out=out)
 
+
+    @ops(filter(reduction_dtype_filter, _ops_and_refs), dtypes=(torch.int16,))
+    def test_out_integral_dtype(self, device, dtype, op):
+        def helper(with_out, expectFail, op_to_test, inputs, *args, **kwargs):
+            out = None
+            try:
+                if with_out:
+                    out = torch.empty(0, dtype=torch.int32)
+                    op_to_test(inputs, out=out, *args, **kwargs)
+                else:
+                    out = op_to_test(inputs, *args, **kwargs)
+                self.assertFalse(expectFail)
+            except RuntimeError:
```

Comment:
don't catch just runtime error, make sure that the error is what you expect

Reply:
Thank you for the review! `RuntimeError` is the error thrown, it can be found here:

https://github.com/pytorch/pytorch/blob/d89dd05e4d25c25d5c95be113a45a6ee61f5d85e/torch/_refs/__init__.py#L2059-L2066

Reply:
I understand that type of the error is "RuntimeError", but you should also check error message, otherwise many errors including those you don't expect will pass.

Reply:
I have added an explicit check for the message.

Reply:
Cool, feel free to land with pytorchbot command when CI looks good.